core/storage/topic/gae_models_test.py

```diff
@@ -0,0 +1,147 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test for Topic model."""
+
+from constants import constants
+from core.domain import subtopic_page_domain
+from core.domain import subtopic_page_services
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+
+
+class TopicModelUnitTest(test_utils.GenericTestBase):
+    """Test the TopicModel class."""
+
+    def setUp(self):
+        super(TopicModelUnitTest, self).setUp()
+        self.topic = topic_domain.Topic.create_default_topic(
```

Comment:
You don't really need to do this in setUp(). Just do it for each test separately.

For example, for the first one, do the creation and then check that the ID of the created topic model is as expected.

For the second one, create a topic, save it, then check that the queries return the correct result.

Reply:
I don't understand. What should I move/change?
If I assign a variable inside a test method then pylint gives me an error that I can't defined variables inside test methods. So I changed it to setUp() method

Reply:
You don't need a setUp() method, since this isn't really the scenario where we have a lot of tests depending on a specific environment that needs to be pre-set-up to avoid massive duplication of code.

That said, I'm a bit concerned to hear about the second point ("If I assign a variable inside a test method then pylint gives me an error that I can't defined variables inside test methods."). Would you mind pasting the error code and the full text of that error message so that I can investigate and provide better advice on this? I don't see the problem with defining variables in general, the issue comes from defining stuff on "self".

Reply:
That's correct. The issue comes from defining stuff on "self", because if I use self then I get this error:
```
W: 37, 8: Attribute 'topic' defined outside __init__ (attribute-defined-outside-init)
W: 45, 8: Attribute 'topic' defined outside __init__ (attribute-defined-outside-init)
```
But if I delete self then there is no error.
So the moved of the assignments are Done in this class and the other class that you mentioned below (commit: https://github.com/oppia/oppia/pull/5689/commits/74f5324cafc14d937b184f68081190bb6c943250).
But, should I moved the variable assignments of the other classes that you didn't mention? Because you only mentioned 2 classes and there were 4 classes with a setUp() method (now 2)

Reply:
Probably yes, but for reference, the general principle is to see if there are lots of tests depending on a specific environment that needs to be pre-set-up to avoid massive duplication of code. If that's not the case then you probably don't need a setUp() method.

Reply:
Well, I removed the remaining setUp() methods as the assignments there were only use once. So, its Done (commit: https://github.com/oppia/oppia/pull/5689/commits/a7bda239d51b9c9c6b9d27f6aed5c3d6eb4494d5)